src/Acuminator/Acuminator.Analyzers/StaticAnalysis/RowChangesInEventHandlers/VariablesWalker.cs

```diff
@@ -84,10 +84,16 @@ public override void VisitIsPatternExpression(IsPatternExpressionSyntax isPatter
 			{
 				_cancellationToken.ThrowIfCancellationRequested();
 
-				if (isPatternExpression.Pattern is DeclarationPatternSyntax declarationPattern && declarationPattern.Designation != null)
+				VariableDesignationSyntax? designation = isPatternExpression.Pattern switch
+				{
+					DeclarationPatternSyntax declarationPattern => declarationPattern.Designation,
+					RecursivePatternSyntax recursivePattern => recursivePattern.Designation,
+					_ => null
+				};
+
+				if (designation != null)
 				{
-					var variableSymbol = _semanticModel.GetDeclaredSymbol(declarationPattern.Designation, _cancellationToken) as ILocalSymbol;
-					ValidateThatVariableIsSetToDacFromEvent(variableSymbol, isPatternExpression.Expression);
+					ValidateThatVariableIsSetToDacFromEvent(_semanticModel.GetDeclaredSymbol(designation, _cancellationToken) as ILocalSymbol, isPatternExpression.Expression);
```

Comment:
@ShadowFiendZX 
Please keep the variable declaration as a separate statement. It's easier to read and maintain simple statements where you can see the result of expression in the variable during the debug
```cs
var variableSymbol = _semanticModel.GetDeclaredSymbol(declarationPattern.Designation, _cancellationToken) as ILocalSymbol;
```

Reply:
Done.